### PR TITLE
fix: correctly propagate sdk.Result events to the current ctx

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

Fixes an issue whereby events would be swallowed into the void due to ignoring of `sdk.Result` fields and more specifically the creation of a new `EventManager` instance within the [`sdk.Msg` router execution](https://github.com/cosmos/cosmos-sdk/blob/v0.44.5/baseapp/msg_service_router.go#L110).

We should also be aware that a new `EventManager` instance is created upon every call to [`ctx.CacheContext`](https://github.com/cosmos/cosmos-sdk/blob/v0.44.5/types/context.go#L258)

Shout out @colin-axner for helping identify the cause of the issue.

I tested the following changes using hermes and the interchain-accounts auth module. Channel handshakes are now correctly picked up by hermes channel workers and completed. 

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Comment by seantking** [comment]
hell ya! @colin-axner @damiannolan

**Review by seantking — APPROVED** [review]

File: modules/apps/27-interchain-accounts/controller/keeper/account.go [hunk 1]
```diff
@@ -32,9 +32,13 @@ func (k Keeper) InitInterchainAccount(ctx sdk.Context, connectionID, counterpart
...
 	msg := channeltypes.NewMsgChannelOpenInit(portID, icatypes.VersionPrefix, channeltypes.ORDERED, []string{connectionID}, icatypes.PortID, icatypes.ModuleName)
 	handler := k.msgRouter.Handler(msg)
-	if _, err := handler(ctx, msg); err != nil {
+
+	res, err := handler(ctx, msg)
+	if err != nil {
 		return err
 	}
 
+	ctx.EventManager().EmitEvents(res.GetEvents())
```
Comment (colin-axner) [inline modules/apps/27-interchain-accounts/controller/keeper/account.go:41]: we should add a comment explaining why this is necessary so it isn't deleted later

File: modules/apps/27-interchain-accounts/host/keeper/relay.go [hunk 2]
```diff
@@ -46,24 +46,32 @@ func (k Keeper) executeTx(ctx sdk.Context, sourcePort, destPort, destChannel str
 			return err
 		}
 
-		if _, err := k.executeMsg(cacheCtx, msg); err != nil {
+		if err := k.executeMsg(cacheCtx, msg); err != nil {
 			return err
 		}
 	}
 
+	ctx.EventManager().EmitEvents(cacheCtx.EventManager().Events())
```
Comment (colin-axner) [inline modules/apps/27-interchain-accounts/host/keeper/relay.go:54]: ditto on comment

File: modules/apps/27-interchain-accounts/host/keeper/relay.go [hunk 3]
```diff
@@ -46,24 +46,32 @@ func (k Keeper) executeTx(ctx sdk.Context, sourcePort, destPort, destChannel str
...
+		return icatypes.ErrInvalidRoute
 	}
 
-	return handler(ctx, msg)
+	res, err := handler(ctx, msg)
+	if err != nil {
+		return err
+	}
+
+	ctx.EventManager().EmitEvents(res.GetEvents())
```
Comment (colin-axner) [inline modules/apps/27-interchain-accounts/host/keeper/relay.go:72]: add comment

**Review by colin-axner — APPROVED** [review]
Great! Thanks for posting the fix! Only request is to add some comments since this is very subtle. In the future we could also add tests to ensure OnChanOpenInit events are returned (but can be done in separate pr)

**Comment by damiannolan** [comment]
> Great! Thanks for posting the fix! Only request is to add some comments since this is very subtle. In the future we could also add tests to ensure OnChanOpenInit events are returned (but can be done in separate pr)

Good point! Updated with some inline comments :)

**Pull request merged**